examples/mongodbatlas_advanced_cluster/asymmetric-sharded-cluster/main.tf

```diff
@@ -0,0 +1,79 @@
+provider "mongodbatlas" {
+  public_key  = var.public_key
+  private_key = var.private_key
+}
+
+resource "mongodbatlas_advanced_cluster" "cluster" {
+  project_id     = mongodbatlas_project.project.id
+  name           = var.cluster_name
+  cluster_type   = "SHARDED"
+  backup_enabled = true
+
+  replication_specs { # shard 1 - M30 instance size
+    region_configs {
+      electable_specs {
+        instance_size = "M30"
+        disk_iops     = 3000
+        node_count    = 3
+      }
+      provider_name = "AWS"
+      priority      = 7
+      region_name   = "EU_WEST_1"
+    }
+  }
+
+  replication_specs { # shard 2 - M20 instance size
+    region_configs {
+      electable_specs {
+        instance_size = "M20"
+        disk_iops     = 3000
+        node_count    = 3
+      }
+      provider_name = "AWS"
+      priority      = 7
+      region_name   = "EU_WEST_1"
+    }
+  }
+
+  replication_specs { # shard 3 - M10 instance size
+    region_configs {
+      electable_specs {
+        instance_size = "M10"
+        disk_iops     = 3000
+        node_count    = 3
+      }
+      provider_name = "AWS"
+      priority      = 7
+      region_name   = "EU_SOUTH_2"
+    }
+  }
+
+  replication_specs { # shard 4 - M10 instance size
+    region_configs {
+      electable_specs {
+        instance_size = "M10"
+        disk_iops     = 3000
+        node_count    = 3
+      }
+      provider_name = "AWS"
+      priority      = 7
+      region_name   = "EU_SOUTH_2"
```

Comment:
was it intentional to leave the first 2 shards with region `EU_WEST_1`, and the other 2 with `EU_SOUTH_2`? I was actually surprised to see this creation work.

Reply:
are there supposed to be region restrictions ?

Reply:
My understanding is that only instance_size and disk_iops can be scaled independently between shards in a zone, would wait for the zone id ticket to be resolved and then I can test out this scenario again.